javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/InstrumentationModuleClassLoader.java

```diff
@@ -167,7 +250,12 @@ public URL getResource(String resourceName) {
     if (fromAgentCl != null) {
       return fromAgentCl;
     }
-    return instrumentedCl.getResource(resourceName);
+
+    if (instrumentedCl != null) {
+      return instrumentedCl.getResource(resourceName);
+    } else {
+      return BOOT_LOADER.getResource(resourceName);
```

Comment:
Don't both agent loader and instrumented cl already delegate to boot loader?

Reply:
For the instrumented CL: We don't know, the Classloader could be doing something weird regarding delegation (e.g. OSGi). 
For the agent loader I think we use a child/self-first loading, so that e.g. if someone puts bytebuddy on the bootstrap it doesn't affect us, right? That's at least what we do in the elastic-apm-agent, because we had that exact problem before. In that case resources from the agent loader might shadow the resource we actually want to lookup here.

That wouldn't exactly be the problem here, because we try the agent loader before anyway and it returned `null` when we reach this code. But if that ever changes, this could easily introduce a subtile bug here. For that reason I'd prefer to keep it explicit here with the `BOOT_LOADER`.

Reply:
> For the instrumented CL: We don't know, the Classloader could be doing something weird regarding delegation (e.g. OSGi).

We instrument class loaders to force boot delegation for packages with our classes to ensure they are accessible with OSGI class loaders.

 > For the agent loader I think we use a child/self-first loading, so that e.g. if someone puts bytebuddy on the bootstrap it doesn't affect us, right?

Yes. There may still be ways to break it, if the libraries we use have optional components that we don't have in agent then placing them in boot loader could produce unexpected results.

> That wouldn't exactly be the problem here, because we try the agent loader before anyway and it returned null when we reach this code. But if that ever changes, this could easily introduce a subtile bug here. For that reason I'd prefer to keep it explicit here with the BOOT_LOADER.

Imo you are overthinking this, if the agent class loader and the instrumented class loader can't find it then it is probably not meant to be found. I think we can keep it for now.